### ISSUE TITLE
Code completion on other characters than tab or return

### ISSUE THREAD

**Issue by mat-mcloughlin** [adobe/brackets#10464]
Hey,

Working my way through a change request to allow other characters to be used instead of tab or return for code completion (See this discussion)

The plan is to have a `completeHintOnOther` array that works in a similar way to the `completeHintOnTab` property. It works well however it has raised one question which is when a character like `.` is used to complete then you more than likely want to insert the `.` as well as the completion text.

Now I can handle this as part of the extension code or I could figure out a way to get it into the brackets code base. Just wondering your thoughts on this?

**Comment by redmunds** [comment]
@mat-mcloughlin I think you forgot to add link to "See this discussion" text, but I think I remember it being discussed.

I can't think of any scenario when I would want to both insert the currently selected hint *and* insert the character I typed. Can you provide an example of when you think this would be useful?

**Comment by mat-mcloughlin** [comment]
Sure. 

In c# most devs are use to it completing on `.`, `(`, `[` and maybe some others, can't remember. 

So, for example,  if you have half the symbol completed and type `.` you would want it to complete and add the `.` ready to type in the property:

`someObj`
type `.`
`someObject.`
then you can just start typing
`someObject.somePro`

Hope this makes sense?

I don't mind covering that case in my extension but are you going to be happy to accept the PR for the `completeOnOther` (If its all correct :) )

**Comment by RaymondLim** [comment]
When I first noticed this request, I was skeptical that user may want to have both hint completion and insertion of a special character. Just try it out with C# code hints in Xamarin and you're absolutely right. It not only accepts the highlighted hint, but also insert the character after the inserted hint.

Regarding your question of whether to implement it in an extension or core, I think this has to be done in core since you need to modify `CodeHintList.prototype.isHandlingKeyCode` and `CodeHintList.prototype._keydownHook` to accept the highlighted hint.

**Comment by mat-mcloughlin** [comment]
@RaymondLim yeah, I've already made the change to do this locally so it will complete on `.`. But in order to get it also insert the `.` character I need to allow the event propagation to continue so I thought I'd check to make sure its ok with you guys before I go ahead

**Comment by RaymondLim** [comment]
It should be ok if you implement it in a similar way as `completeHintOnTab` with a pref that defaults to `false`. i.e. Nothing really changes for current Brackets users by default, but anyone who wants `completeHintOnOther` can turn it on with the new pref.

**Comment by RaymondLim** [comment]
@mat-mcloughlin I wonder how you're going to provide the list of characters for each language from each hint provider since `.([` characters do not apply to html hints or css hints.

**Comment by mat-mcloughlin** [comment]
I figured I'd allow it to only be set whilst creating a new `CodeHintManager`
